### PR TITLE
Feature/more docs 2

### PR DESCRIPTION
**Pull request by zicklag**
Adds the next chapter to the guide.

### CODE REVIEW

**Comment by zicklag** [comment]
Check it out! Website doc tests are running in CI now, too!

image

File: site/content/docs/getting-started/03-containers/index.md
Comment (PsichiX) [inline site/content/docs/getting-started/03-containers/index.md:23]: we should add a note that transform is used generally by the renderers only and it doesn't affect the layout areas produced by default layout engine, which means transforms are ignored also by interactions (pointer positions) - in other words: transforms are purely visual properties so user should be careful when setting them since they affect all the children visuals also.
i use them in Oxygengine to get effects like scaling or skewing of images, menu screens animated slide in/out, etc.

File: site/content/docs/getting-started/03-containers/index.md
Comment (PsichiX) [inline site/content/docs/getting-started/03-containers/index.md:153]: typo i believe:
Notice that the content box didn't make sure **_there_** was any "room" for the text

File: site/content/docs/getting-started/03-containers/index.md
Comment (PsichiX) [inline site/content/docs/getting-started/03-containers/index.md:161]: the thing with flex box in RAUI is that it might seems to be exact the same as CSS flexbox but it was purposefully modified. After the review i'm going to investigate the margin and grow/shrink problem you've found in the other issue, if they are bugs they are gonna be fixed, but RAUI flex box won't be exactly same as CSS flex box bc RAUI flex box is focused on single pass size calculations which brings some edge cases but doesn't produce performance problems with multipass box size calculations as solutions that strictly follow CSS flex box - after the issue investigation i'll write a description of RAUI version of flex box and it's rules to clarify misconceptions (this is also the case where now i honestly regret calling it a flex box, because it might bring wrong assumptions 🤔) so expect some commits from me to this PR :p

**Review by PsichiX — COMMENTED** [review]
found some type and left a note about flex box layouting that i'll have to clarify in a new commit, all the other changes looks good to me!

File: site/content/docs/getting-started/03-containers/index.md
Comment (zicklag) [reply site/content/docs/getting-started/03-containers/index.md:23]: Oh, OK. I didn't know that. Definitely a good tip to have in there! I'll update that.

File: site/content/docs/getting-started/03-containers/index.md
Comment (zicklag) [reply site/content/docs/getting-started/03-containers/index.md:161]: That makes sense. I wasn't sure how much it was intended to match CSS or not so I didn't want to say it shoud be pretty close if it wasn't supposed to be, but I'd feel comfortable saying that it should be the same in most situations in that case.

> so expect some commits from me to this PR :D

I actually opened the PR from my organization account so it doesn't have that "allow maintainers to push commits" feature. You could create a separate branch that I could merge into this branch, or we could just merge your changes after  merging this PR maybe?

Comment (PsichiX) [reply site/content/docs/getting-started/03-containers/index.md:161]: > or we could just merge your changes after merging this PR maybe?
that works for me!

Comment (zicklag) [reply site/content/docs/getting-started/03-containers/index.md]: I updated the comment just to say that they are similar to CSS Flexboxes, and added a note that links to a new ( blank ) "Layout in Depth" page. You can use that section to write about how layout works in more detail. Feel free to rename the page to whatever you want.

I think that means we can merge this and then you can push whatever commits extra you want to `next` for the guide.

**Review by PsichiX — APPROVED** [review]
everything looks ok, approved!
also, big thanks for preparing a page for my explanation of default layout engine <3 gonna write it today or tomorrow :D

**Review by ItsYoutacRRS — COMMENTED** [review]
#52 #32 #25 #72 #25 #70

**Pull request merged**
